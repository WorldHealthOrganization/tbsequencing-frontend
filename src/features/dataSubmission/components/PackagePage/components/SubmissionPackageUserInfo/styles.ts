import { css } from '@emotion/react';
import appColors from '../../../../../../styles/colors';

export const section = css({
  marginTop: 24,
  marginBottom: 24,
});

export const sectionInfo = css({
  marginTop: 8,
  fontSize: '16px',
});

export const headerWrapper = css({
  display: 'flex',
});

const VERIFICATION_IN_PROGRESS_BADGE_COLOR = appColors.secondary.whoMagenta;
const APPROVED_BADGE_COLOR = appColors.secondary.whoGreen;

export const verificationInProgressBadge = css({
  color: VERIFICATION_IN_PROGRESS_BADGE_COLOR,
  border: `1px solid ${VERIFICATION_IN_PROGRESS_BADGE_COLOR}`,
  backgroundColor: appColors.neutral.white,
});

export const approvedBadge = css({
  color: appColors.neutral.white,
  backgroundColor: APPROVED_BADGE_COLOR,
  borderColor: APPROVED_BADGE_COLOR,
});
