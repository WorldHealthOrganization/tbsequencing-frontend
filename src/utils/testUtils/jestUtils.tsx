import React, { ReactElement } from 'react';
import { render as rtlRender } from '@testing-library/react';
import { configureStore } from '@reduxjs/toolkit';
import { Provider } from 'react-redux';

import renderer from 'react-test-renderer';
import { BrowserRouter } from 'react-router-dom';
import { reducer } from '../../app/store';

const getWrapper = (
  ui: ReactElement,
  {
    // @ts-ignore
    preloadedState,
    store = configureStore({ reducer, preloadedState }),
  } = {},
) => ({ children }: { children: ReactElement }) => (
  <Provider store={store}>{children}</Provider>
);

function render(
  ui: ReactElement,
  {
    // @ts-ignore
    preloadedState,
    store = configureStore({ reducer, preloadedState }),
    ...renderOptions
  } = {},
) {
  const Wrapper = getWrapper(ui, { preloadedState, store });

  return rtlRender(ui, { wrapper: Wrapper, ...renderOptions });
}

const getTreeJSONWithRedux = (ui: ReactElement, {
  // @ts-ignore
  preloadedState,
  store = configureStore({ reducer, preloadedState }),
} = {}) => {
  const Wrapper = getWrapper(ui, { preloadedState, store });

  const tree = renderer
    .create(<Wrapper>{ui}</Wrapper>)
    .toJSON();

  return tree;
};

export const wrappedInRouter = (
  component: ReactElement,
) => <BrowserRouter>{component}</BrowserRouter>;

export * from '@testing-library/react';
export { render, getTreeJSONWithRedux };
