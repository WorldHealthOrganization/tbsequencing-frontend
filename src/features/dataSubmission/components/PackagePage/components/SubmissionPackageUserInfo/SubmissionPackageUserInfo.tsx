/** @jsxImportSource @emotion/react */
import React from 'react';
import { SerializedStyles } from '@emotion/react';
import { useMsal } from '@azure/msal-react';
import * as styles from './styles';
import AppPaper from '../../../../../../components/AppPaper';
import H2 from '../../../../../../components/typography/H2';
import H3 from '../../../../../../components/typography/H3';
import PrimaryText from '../../../../../../components/typography/PrimaryText';

interface Props {
  containerStyle?: SerializedStyles;
}

export const SubmissionPackageUserInfo = ({
  containerStyle,
}:
Props) => {
  const { accounts } = useMsal();

  if (!accounts[0]) {
    return null;
  }

  return (
    <AppPaper style={containerStyle}>
      <div css={styles.headerWrapper}>
        <H2>General info</H2>
      </div>
      <section css={styles.section}>
        <PrimaryText style={styles.sectionInfo}>{accounts[0].name}</PrimaryText>
      </section>
      {accounts[0].idTokenClaims?.email !== undefined && (
        <section css={styles.section}>
          <H3>Email</H3>
          <PrimaryText style={styles.sectionInfo}>
            {accounts[0].idTokenClaims?.email as string}
          </PrimaryText>
        </section>
      )}
    </AppPaper>
  );
};

export default SubmissionPackageUserInfo;
