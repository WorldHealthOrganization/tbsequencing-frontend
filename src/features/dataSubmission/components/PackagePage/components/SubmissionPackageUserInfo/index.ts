import SubmissionPackageUserInfo from './SubmissionPackageUserInfo';

export default SubmissionPackageUserInfo;
