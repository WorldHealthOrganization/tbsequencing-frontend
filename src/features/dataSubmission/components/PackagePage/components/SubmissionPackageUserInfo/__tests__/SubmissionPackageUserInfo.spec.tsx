import React from 'react';
import { getTreeJSONWithRedux, render } from '../../../../../../../utils/testUtils/jestUtils';
import SubmissionPackageUserInfo from '../SubmissionPackageUserInfo';

// TODO add more tests and mock useUser somehow

describe('SubmissionPackageUserInfo component test', () => {
  const onChangeInfoMockHandler = jest.fn();

  test('renders SubmissionPackageUserInfo component without error', () => {
    render(
      <SubmissionPackageUserInfo
        onChangeInfoClick={onChangeInfoMockHandler}
      />,
    );
  });

  it('Should render correctly', () => {
    const tree = getTreeJSONWithRedux(
      <SubmissionPackageUserInfo
        onChangeInfoClick={onChangeInfoMockHandler}
      />,
    );

    expect(tree).toMatchSnapshot();
  });
});
