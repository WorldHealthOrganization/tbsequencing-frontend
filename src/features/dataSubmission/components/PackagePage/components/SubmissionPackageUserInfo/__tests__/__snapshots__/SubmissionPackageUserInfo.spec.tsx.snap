// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SubmissionPackageUserInfo component test Should render correctly 1`] = `null`;
